Calendars/Calendars.Plugin.Tizen/CalendarsImplementation.cs

```diff
@@ -0,0 +1,438 @@
+						AccountName = calRecord.Get<int>(Book.AccountId).ToString(),
+						Color = calRecord.Get<string>(Book.Color),
+					}
+				);
+			}
+			finally
+			{
+				calRecord?.Dispose();
+				calRecord = null;
+			}
+			return Task.FromResult<IList<Calendar>>(calendars);
+		}
+
+		/// <summary>
+		/// Gets a single calendar by platform-specific ID.
+		/// </summary>
+		/// <param name="externalId">Platform-specific calendar identifier</param>
+		/// <returns>The corresponding calendar, or null if not found</returns>
+		/// <exception cref="System.UnauthorizedAccessException">Calendar access denied</exception>
+		/// <exception cref="Plugin.Calendars.Abstractions.PlatformException">Unexpected platform-specific error</exception>
+		public Task<Calendar> GetCalendarByIdAsync(string externalId)
+		{
+			Calendar calendar = null;
+			CalendarRecord calRecord = null;
+
+			try
+			{
+				calendar = new Calendar();
+				calRecord = new CalendarRecord(Book.Uri);
+				calRecord.Set<string>(Book.Name, "org.tizen.calendar");
+				calendar.Name = calRecord.Get<string>(Book.Name);
+				calendar.ExternalID = calRecord.Get<int>(Book.Id).ToString();
+				calendar.CanEditEvents = true;
+				calendar.CanEditCalendar = false;
+				calendar.AccountName = calRecord.Get<int>(Book.AccountId).ToString();
+				calendar.Color = calRecord.Get<string>(Book.Color);
+			}
+			finally
+			{
+				calRecord?.Dispose();
+				calRecord = null;
+			}
+			return Task.FromResult(calendar);
+		}
+
+		/// <summary>
+		/// Gets all events for a calendar within the specified time range.
+		/// </summary>
+		/// <param name="calendar">Calendar containing events</param>
+		/// <param name="start">Start of event range</param>
+		/// <param name="end">End of event range</param>
+		/// <returns>Calendar events</returns>
+		/// <exception cref="System.ArgumentException">Calendar does not exist on device</exception>
+		/// <exception cref="System.UnauthorizedAccessException">Calendar access denied</exception>
+		/// <exception cref="Plugin.Calendars.Abstractions.PlatformException">Unexpected platform-specific error</exception>
+		public Task<IList<CalendarEvent>> GetEventsAsync(Calendar calendar, DateTime start, DateTime end)
+		{
+			var calEvent = new List<CalendarEvent>();
+			CalendarManager calManager = null;
+			CalendarQuery calQuery = null;
+			CalendarList calList = null;
+			CalendarRecord calRecord = null;
+			CalendarFilter calFilter = null;
+
+			try
+			{
+				calManager = new CalendarManager();
+				calFilter.AddCondition(CalendarFilter.LogicalOperator.And, Event.Start, CalendarFilter.IntegerMatchType.GreaterThanOrEqual, ConvertIntPtrToCalendarTime(start));
+				calFilter.AddCondition(CalendarFilter.LogicalOperator.And, Event.End, CalendarFilter.IntegerMatchType.LessThanOrEqual, ConvertIntPtrToCalendarTime(end));
+				calQuery = new CalendarQuery(Event.Uri);
+				calQuery.SetFilter(calFilter);
+				calList = calManager.Database.GetRecordsWithQuery(calQuery, 0, 0);
```

Comment:
So, I know I suggested this API based on the Tizen docs... but when trying it, I was surprised to see that it was missing a lot of events...

...and then I noticed that the docs specifically say in the example:
> ...to filter _**all-day events**_ that start after January 1st, 2016...

And, indeed, it returns all-day events, but not events that have specific times. And I can't find anything in the Tizen docs about how one would query for non-all-day events. Maybe it's just not supported??? Which seems very strange, but I guess would mean having to go back to `GetAll`.......